### PR TITLE
Fixed the build issues from nuget and msbuild.

### LINKED ISSUE #3711: nuget and msbuild errors when creating building the Dockerfile.
#### Expected Behavior

Build without any nuget or msbuild issue.

#### Actual Behavior

I get the following error
```
The 'Newtonsoft.Json 10.0.3' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.8.60717.93'.
'QLNet' already has a dependency defined for 'System.Reflection.Emit.Lightweight'.
The 'System.Net.Http 4.3.3' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.8.60717.93'.
The 'Microsoft.CSharp 4.6.0' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.8.60717.93'.
The 'System.Reflection.Emit 4.3.0' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.8.60717.93'.
The 'System.Reflection.Emit.Lightweight 4.3.0' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.8.60717.93'.
The 'System.ValueTuple 4.4.0' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.8.60717.93'.
```

After upgrading the nuget version, I get the following msbuild error
```
MSBUILD : error MSB1011: Specify which project or solution file to use because this folder contains more than one project or solution file.
```

#### Potential Solution

Downloading latest nuget and specif
#### Reproducing the Problem

Upgrade the Nuget version then specify the solutions file when calling msbuild.

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by randomthought**
#### Description

Fixed build issues with nuget and msbuild when creating a docker image.

#### Related Issue

Closes #3711

#### Motivation and Context

When using docker, a build should always work unless the Dockerfile hasn't been kept up to date with build steps.

#### Requires Documentation Change

N/A

#### How Has This Been Tested?

No need to create a test case for this. It either builds or it doesn't.

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,9 +16,9 @@ MAINTAINER QuantConnect <contact@quantconnect.com>
 # RUN \
 #     cd /root/Lean && \
 #     sed -i 's/4.5/4.0/' Algorithm.VisualBasic/QuantConnect.Algorithm.VisualBasic.vbproj && \
-#     wget https://nuget.org/nuget.exe && \
+#     wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe && \
```
Comment (mchandschuh) [inline Dockerfile]: The old link is working fine for me. Why was this change required/

File: Dockerfile [hunk 2]
```diff
@@ -16,9 +16,9 @@ MAINTAINER QuantConnect <contact@quantconnect.com>
...
 #     mono nuget.exe restore QuantConnect.Lean.sln -NonInteractive && \
-#     msbuild /property:Configuration=Release && \
+#     msbuild QuantConnect.Lean.sln /property:Configuration=Release && \
```
Comment (mchandschuh) [inline Dockerfile]: I'm surprised this wasn't working either? What was the error resolved by this change?

File: Dockerfile (hunk 1 above)
Comment (mchandschuh) [reply Dockerfile]: The error provided in the linked issue complains about nuget packages requiring version `2.1.2` and I've confirmed that the originally linked nuget installer is for `2.8.61717.93`.

Comment (randomthought) [reply Dockerfile]: The errors in the #3711 states it requires `2.12` or above not `2.1.2` :)

File: Dockerfile (hunk 2 above)
Comment (randomthought) [reply Dockerfile]: I described the error in #3711, posting it here for reference.
```
MSBUILD : error MSB1011: Specify which project or solution file to use because this folder contains more than one project or solution file.
```

**Comment by randomthought** [comment]
> Posted a couple comments to gather more information about what was failing and why these changes fixed it as this wasn't covered in the PR description _or_ the commit message.

Sorry you had to go through the trouble due to the lack of explanation on this PR. I usually create an issue describing in detail what is the problem before creating a PR to resolve the issue. The PR isn't as descriptive since the issue has all the content (although I should have been more descriptive with my issue in the first place).

File: Dockerfile (hunk 1 above)
Comment (mchandschuh) [reply Dockerfile]: wow, further, nuget is _far_ beyond major version 2 and at about `5.3.x`
